### DISCUSSION TITLE
ERR_INVALID_ARG_TYPE when reading from S3

### DISCUSSION THREAD

**Discussion by tedyu** [aws/aws-sdk-js#4668] [Q&A]
We face the following exception using sdk v2 to read data from s3:
```
message:The "input" argument must be an instance of ArrayBuffer or ArrayBufferView. Received type number (123)
stack:TypeError [ERR_INVALID_ARG_TYPE]: The "input" argument must be an instance of ArrayBuffer or ArrayBufferView. Received type number (123)
at new NodeError (node:internal/errors:387:5)
at TextDecoder.decode (node:internal/encoding:417:15)
```
Here is related code:
```
    EventHelper.getS3()
      .getObject({
        Key: key,
        Bucket: bucket || EventHelper.getDefaultEventsBucket(),
      }, async (err, res) => {
        if (err) { return callback(err); }
        let streamErr;
        const raw = await consumers.text(res.Body).catch((err2) => {
```
`consumers.text` is where the exception comes.

If the writer to s3 uses sdk v3, I assume the interaction between v3 writer and v2 reader should work.
v3 ver: 3.50.0
v2 ver: 2.1341.0

Has anyone faced similar exception ?

**Comment by aBurmeseDev** [comment]
Hi @tedyu - as mentioned in [the post](https://github.com/aws/aws-sdk-js-v3/discussions/6399) you created in aws-sdk-js-v3 repo, the problem you're facing is related to how you're trying to consume the response stream from the S3 object and you'll need to consume the response stream. Here are how you can resolve in v2:

```js
EventHelper.getS3()
  .getObject({
    Key: key,
    Bucket: bucket || EventHelper.getDefaultEventsBucket(),
  }, (err, data) => {
    if (err) {
      return callback(err);
    }

    const rawData = data.Body.toString('utf-8');
    // Process rawData as needed
    callback(null, rawData);
  });
```
Best,
John

**Comment by tedyu** [comment]
@aBurmeseDev 
Thanks for the response.
In our case, we received number when reading S3. Just want to see if you have seen such scenario.
According to your suggestion, it seems we should convert the number into string first.

**Reply by aBurmeseDev** [reply]
If you continue to experience issues after attempting to consume the response stream and convert it to a string, please let me know. We can further investigate the problem and explore alternative solutions or workarounds.
